onnxruntime/core/session/ep_library_provider_bridge.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/session/ep_library_provider_bridge.h"
+
+#include "core/framework/session_options.h"
+#include "core/providers/cuda/cuda_provider_options.h"
+#include "core/providers/shared_library/provider_host_api.h"
+#include "core/session/abi_devices.h"
+#include "core/session/abi_session_options_impl.h"
+#include "core/session/ep_factory_internal.h"
+
+namespace onnxruntime {
+
+std::unique_ptr<EpFactoryInternal> EpLibraryProviderBridge::CreateCudaEpFactory(Provider& provider) {
+  // Using the name that SessionOptionsAppendExecutionProvider uses to identify the EP as that matches the
+  // expected name in the configuration options. must be static to be valid for the lambdas
+  static const std::string ep_name = "CUDA";
```

Comment:
Is it worth adding the 'is supported' and 'create ep' implementation to the library to avoid hardcoding things on the ORT time?

Reply:
I think so. See #24455 and if that approach seems fine this hardcoding will go away.